File: Leetcode/Monthly Challenges/November_LeetCoding_Challenge/ConsecutiveCharacters/ConsecutiveCharacters.cpp
class Solution {
public:
    int maxPower(string s) {
        if(s.size() == 1)
            return 1;
        
        int max_count = 1;
        int count = 1;
        for(int i=0; i<s.size(); i++)
        {
            if(s[i] == s[i+1])
                count++;
            if(count > max_count)
                max_count = count;
            if(s[i] != s[i+1])
                count = 1;
        }
        return max_count;
    }
};
// Time Complexity - O(N), N is the number of characters in the string
// Space Complexity - O(1), constant space

File: LeetCode/Monthly Challenges/November_LeetCoding_Challenge/ConsecutiveCharacters/ConsecutiveCharacters.py
class Solution:
    def maxPower(self, s: str) -> int:
        max_count = 1
        count = 1
        for i in range(len(s)-1):
            if(s[i] == s[i+1]):
                count += 1
            if(count > max_count):
                max_count = count
            if(s[i] != s[i+1]):
                count = 1
        return max_count

# Time Complexity - O(n)
# Space Complexity - O(1) -> Linear Space
